### PR TITLE
test: check uv_cond_timedwait more carefully

### PR DESCRIPTION
**Pull request by davisjam**
Problem:
The "timeout" functionality of uv_cond_timedwait was not being tested.
The test (condvar_3) would use a worker that signaled the condition.

Solution:
Introduce a new condvar test case to ensure that the timeout also works.

### CODE REVIEW

**Comment by davisjam** [comment]
@jBarz Thoughts, since you were looking at this recently?

File: test/test-condvar.c [hunk 1]
```diff
@@ -120,20 +125,24 @@ TEST_IMPL(condvar_2) {
 }
 
 
-static void condvar_timedwait(worker_config* c, const volatile int* flag) {
-  int r;
+static int condvar_timedwait(worker_config* c, const volatile int* flag) {
+  int r = 0;
```
Comment (bnoordhuis) [inline test/test-condvar.c]: Tiniest of nits but libuv usually separates declaration from assignment.  I.e.:
```c
int r;

r = 0;
```

File: test/test-condvar.c [hunk 2]
```diff
@@ -120,20 +125,24 @@ TEST_IMPL(condvar_2) {
...
 
   uv_mutex_lock(&c->mutex);
   if (c->wait_delay)
     uv_sleep(c->wait_delay);
   while (*flag == 0) {
     r = uv_cond_timedwait(&c->cond, &c->mutex, (uint64_t)(150 * 1e6));
-    ASSERT(r == 0);
+    ASSERT(r == 0 || r == -ETIMEDOUT);
```
Comment (bnoordhuis) [inline test/test-condvar.c]: UV_ETIMEDOUT

File: test/test-condvar.c [hunk 3]
```diff
@@ -205,3 +214,30 @@ TEST_IMPL(condvar_5) {
...
+  wc.wait_cond = condvar_timedwait;
+
+  ASSERT(0 == uv_cond_init(&wc.cond));
+  ASSERT(0 == uv_mutex_init(&wc.mutex));
+  ASSERT(0 == uv_thread_create(&thread, noop_worker, &wc));
+
+  /* This can only return having timed out, because otherwise we
+   * loop forever in condvar_timedwait. */
+  r = wc.wait_cond(&wc, &wc.posted_1);
+  ASSERT(r == -ETIMEDOUT);
```
Comment (bnoordhuis) [inline test/test-condvar.c]: Ditto.

**Comment by jBarz** [comment]
LGTM

**Comment by bnoordhuis** [comment]
https://ci.nodejs.org/job/libuv-test-commit/696/

**Comment by bnoordhuis** [comment]
Thanks Jamie, landed in 693c217f8.

**Pull request closed without merging**